liquibase-standard/src/main/java/liquibase/util/StandardSqlParser.java

```diff
@@ -48,17 +53,52 @@ public StringClauses parse(String sqlBlock, boolean preserveWhitespace, boolean
                 }
                 token = t.getNextToken();
             }
-
         } catch (Throwable e) {
             if (changeSet != null) {
-                throw new UnexpectedLiquibaseException(changeSet.toString(), e);
+                Change change = Scope.getCurrentScope().get(ChangeSet.CHANGE_KEY, Change.class);
+                String message = enhanceExceptionMessage(changeSet, change, e);
```

Comment:
I think the call to enhanceExceptionMessage should itself be inside a try catch. We don't want to fail on the exception processing while trying to enhance the message. If we can't enhance the message, just use the existing message.

Reply:
I agree.